packages/babel-plugin-transform-class-properties/src/index.js

```diff
@@ -96,6 +96,18 @@ export default function({ types: t }, options) {
           if (isStatic) {
             nodes.push(buildClassProperty(ref, propNode, path.scope));
           } else {
+            // Make sure computed property names are only evaluated once (upon
+            // class definition).
+            if (propNode.computed) {
+              const ident = path.scope.generateUidIdentifier("computed");
```

Comment:
Here you could use `path.scope.generateUidIdentifierBasedOnNode(propNode.value, "computed")`, which generates better names.

e.g. `class A { [x.y] = 2; }` should become something like:
```js
var A = function A() {
  _classCallCheck(this, A);

  Object.defineProperty(this, _x$y, {
    configurable: true,
    enumerable: true,
    writable: true,
    value: 2
  });
};

var _x$y = x;
```

Reply:
Ah, didn't know about that. Thanks!